src/components/button/styled.tsx

```diff
@@ -126,18 +126,10 @@ export const StyledButtonWrapper = styled.div<{ label: string }>`
   }};
 `
 
-export const StyledIcon = styled(({ name }) => <Icon className="button-icon" name={name} />)`
-  ${props => {
-    return css`
-      fill: ${buttonProps("buttonTextColor", props)};
-    `
-  }};
-`
-
-export const StyledButton = styled(({ label, icon, ...otherProps }) => (
+export const StyledButton = styled(({ label, icon, danger, ...otherProps }) => (
   <StyledButtonWrapper label={label}>
-    <button type="button" {...otherProps}>
-      {icon ? <StyledIcon name={icon} {...otherProps} /> : null}
+    <button type="button" danger={danger.toString()} {...otherProps}>
+      {icon ? <Icon className="button-icon" name={icon} /> : null}
```

Comment:
What is the specific reason to use classname?

Reply:
This is how this component is selected in the CSS